BuildScript/BuildTasks/TestConfigurationsFilter.cs

```diff
@@ -0,0 +1,201 @@
+      _enabledDatabaseSystems = enabledDatabaseSystems;
+      _supportedPlatforms = supportedPlatforms;
+      _enabledPlatforms = enabledPlatforms;
+      _supportedExecutionRuntimes = supportedExecutionRuntimes;
+      _enabledExecutionRuntimes = enabledExecutionRuntimes;
+      _supportedTargetRuntimes = supportedTargetRuntimes;
+      _enabledTargetRuntimes = enabledTargetRuntimes;
+      _supportedConfigurationIDs = supportedConfigurationIDs;
+      _enabledConfigurationIDs = enabledConfigurationIDs;
+    }
+
+    public IReadOnlyCollection<TestConfiguration> GetFilteredTestConfigurations (IReadOnlyCollection<TestConfiguration> testConfigurationsToFilter)
+    {
+      var errorsForUnsupportedMetadata = GetErrorsForUnsupportedMetadata (testConfigurationsToFilter).ToArray();
+      if (errorsForUnsupportedMetadata.Any())
+        throw new InvalidOperationException (string.Join (Environment.NewLine, errorsForUnsupportedMetadata));
+
+      return testConfigurationsToFilter
+          .Where (HasEnabledBrowser)
+          .Where (HasEnabledPlatform)
+          .Where (HasEnabledDatabaseSystem)
+          .Where (HasEnabledExecutionRuntime)
+          .Where (HasEnabledTargetRuntime)
+          .Where (HasEnabledConfigurationID)
+          .ToArray();
+    }
+
+    private IEnumerable<string> GetErrorsForUnsupportedMetadata (IEnumerable<TestConfiguration> testConfigurations)
+    {
+      foreach (var testConfiguration in testConfigurations)
+      {
+        if (!IsSupportedBrowser (testConfiguration.Browser))
+          yield return GetErrorMessageForUnsupportedFilterValue (TestConfigurationMetadata.Browser, testConfiguration.Browser);
+
+        if (!IsSupportedPlatform (testConfiguration.Platform))
+          yield return GetErrorMessageForUnsupportedFilterValue (TestConfigurationMetadata.Platform, testConfiguration.Platform);
+
+        if (!IsSupportedDatabaseSystem (testConfiguration.DatabaseSystem))
+          yield return GetErrorMessageForUnsupportedFilterValue (TestConfigurationMetadata.DatabaseSystem, testConfiguration.DatabaseSystem);
+
+        if (!IsSupportedExecutionRuntime (testConfiguration.ExecutionRuntime))
+          yield return GetErrorMessageForUnsupportedFilterValue (TestConfigurationMetadata.ExecutionRuntime, testConfiguration.ExecutionRuntime);
+
+        if (!IsSupportedTargetRuntime (testConfiguration.TargetRuntime))
+          yield return GetErrorMessageForUnsupportedFilterValue (TestConfigurationMetadata.TargetRuntime, testConfiguration.TargetRuntime);
+
+        if (!IsSupportedConfigurationID (testConfiguration.ConfigurationID))
+          yield return GetErrorMessageForUnsupportedFilterValue (TestConfigurationMetadata.ConfigurationID, testConfiguration.TargetRuntime);
+      }
+    }
+
+    private string GetErrorMessageForUnsupportedFilterValue (string metadata, string value)
+    {
+      return $"Cannot filter with unsupported {metadata} '{value}'";
```

Comment:
quotes. also, see other examples where I check for missing metadata for example text.